### PR TITLE
Tbolt/3469 check valid affiliation year

### LINKED ISSUE #3469: [Feature] Verify state admin expiration date upon login
| Given | When | Then |
| ----- | ---- | ---- |
| a fed admin | logs in with an expired affiliation | is denied access to the respective state |

### This task is done when...

- [ ] State Admins are shown the "revoked" status when logging in with an expired affiliation

Technical implementation details
* When user is logging in, check the expiration date. 
* If expiration date is in the past, update the corresponding affiliation status to revoked
* Ensure JWT is using this latest update to the affiliation status 

---

Add additional labels (design, dev, compliance, BUG, etc) and size before submitting.

If the issue is needed to complete prioritized work for the CURRENT SPRINT, add it to the "This Sprint" pipeline. Otherwise, all other issues will be automatically added to the _unprioritized pipeline_ for prioritization in backlog refinement or sprint planning with Product.

### PR DESCRIPTION
**Pull request by tbolt**
Resolves #3469 

**Description-**
* Verifies affiliation expiration date upon logging in. If it is expired, the affiliation will be set to revoked. 

**This pull request was tested in the follow ways…**

- list of frontend cases
- list of backend cases

**Steps to manually verify this change...**

1. steps to view and verify change

### This pull request is ready to review when...

- [ ] Automated tests are updated (and all tests are passing)
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Review by Sun-Mountain — APPROVED** [review]
CONFIRMED:

| Given         | When     | Then   | Result     | 
|--------------|-----------|------------|------------|
| the user downloads a letter | user clicks on `pdf view` icon | the letter downloads and opens as expected | ✅  PASS|
| the letter downloads | user clicks on `pdf view` icon | the names match the proposed scheme on ticket | ✅  PASS|

File: api/auth/jwtUtils.js [hunk 1]
```diff
@@ -127,16 +131,35 @@ const changeState = async (
...
 ) => {
+  // First verify affiliation is not expired
+  const stateAffiliation = await getAffiliationByState_(user.id, stateId);
+  
+  if (isPast(stateAffiliation.expires_at)) {
+    await updateAuthAffiliation_({
+      affiliationId: stateAffiliation.id,
+      newRoleId: -1,
+      newStatus: 'revoked',
+      changedBy: null,
```
Comment (thetif) [inline api/auth/jwtUtils.js:148]: why don't we change this to 'system' or 'expired'

**Review by thetif — APPROVED** [review]

**Comment by tbolt** [comment]
Update 12/20- @thetif noted that this does not work for users with single affiliations. I'll work to update it for that case.

**Review by thetif — APPROVED** [review]
LGTM

**Comment by thetif** [comment]
Tested by:

|Given|When|Then|Result|
|---|---|---|---|
|a user with one affiliation|the affiliation has not expired|the user has access|:white_check_mark: PASS|
|a user with one affiliation|the affiliation is expired|their access is revoked|:white_check_mark:  PASS|
|a user with multiple affiliations|both affiliations are not expired|the user has access to all affiliations|:white_check_mark: PASS|
|a user with multiple affiliations|one affiliation is expired|the user still has access to the not expired affiliation and the status of the expired affiliation is revoked|:white_check_mark: PASS|

**Pull request merged**